cmd/deltaCmdNode.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/urfave/cli/v2"
+)
+
+var CLIConnectFlags = []cli.Flag{
+	&cli.StringFlag{
+		Name:    "delta-api",
+		Usage:   "API connection info",
+		EnvVars: []string{"DELTA_API"},
+		Hidden:  true,
+	},
+	&cli.StringFlag{
+		Name:    "delta-auth",
+		Usage:   "delta auth token",
+		EnvVars: []string{"DELTA_AUTH"},
+		Hidden:  true,
+	},
+}
+
+type DeltaCmdNode struct {
+	DeltaApi  string
+	DeltaAuth string
+}
+
+func NewDeltaCmdNode(c *cli.Context) (*DeltaCmdNode, error) {
+	deltaApi := c.String("delta-api")
+	if deltaApi == "" {
+		deltaApi = os.Getenv("DELTA_API")
+		if deltaApi == "" {
+			deltaApi = "http://localhost:1414"
+		}
+	}
+
+	deltaAuth := c.String("delta-auth")
+	if deltaAuth == "" {
+		deltaAuth = os.Getenv("DELTA_AUTH")
+		if deltaAuth == "" {
+			return nil, fmt.Errorf("DELTA_AUTH env variable or --delta-auth flag is required")
+		}
+	}
```

Comment:
i'm not a fan of this logic - wonder if there's a cleaner way of doing this?

essentially - take the value from cli - if undefined, take the value from env. if undefined - use default, or error.

Reply:
I created a `getFlagOrEnvVar` function that will wrap the repetitive value check.

```
func getFlagOrEnvVar(c *cli.Context, flagName, envVarName, defaultValue string) string {
	value := c.String(flagName)
	if value == "" {
		value = os.Getenv(envVarName)
		if value == "" {
			value = defaultValue
		}
	}
	return value
}
```

I then call it from the node initialization

```
deltaApi := getFlagOrEnvVar(c, "delta-api", "DELTA_API", "http://localhost:1414")
deltaAuth := getFlagOrEnvVar(c, "delta-auth", "DELTA_AUTH", "")

```